static/js/containers/ChannelFollowControls.js

```diff
@@ -0,0 +1,121 @@
+/* global SETTINGS: false */
+// @flow
+import React from "react"
+import R from "ramda"
+import { connect } from "react-redux"
+
+import DropdownMenu from "../components/DropdownMenu"
+
+import { actions } from "../actions"
+import { hideDropdown, showDropdown } from "../actions/ui"
+import { leaveChannel } from "../util/api_actions"
+
+import type { Dispatch } from "redux"
+import type { Channel } from "../flow/discussionTypes"
+
+export const CHANNEL_FOLLOW_DROPDOWN = "CHANNEL_FOLLOW_DROPDOWN"
+
+type Props = {
+  channel: Channel,
+  isDropdownOpen: boolean,
+  showDropdown: () => void,
+  hideDropdown: () => void,
+  followChannel: () => void,
+  unfollowChannel: () => void,
+  leaveChannel: () => void,
+  loadChannel: () => void
+}
+
+export const DropDownArrow = () => (
+  <i className="material-icons arrow_drop_down">arrow_drop_down</i>
+)
+
+export const DropUpArrow = () => (
+  <i className="material-icons arrow_drop_up">arrow_drop_up</i>
+)
+
+export class ChannelFollowControls extends React.Component<Props> {
+  handleFollowClick = async () => {
+    const { followChannel, loadChannel } = this.props
+
+    await followChannel()
+    await loadChannel()
+  }
+
+  handleUnfollowClick = async () => {
+    const { unfollowChannel, loadChannel } = this.props
+
+    await unfollowChannel()
+    await loadChannel()
+  }
+
+  handleLeaveChannelClick = async () => {
+    const { leaveChannel, loadChannel } = this.props
+
+    await leaveChannel()
+    await loadChannel()
+  }
+
+  render() {
+    const { channel, showDropdown, hideDropdown, isDropdownOpen } = this.props
+
+    return channel.user_is_subscriber ? (
```

Comment:
It's possible for the user to be a contributor but not a subscriber, which means this logic wouldn't allow them to leave the channel without re-subscribing first.

Reply:
I'm going to call this one out of scope. The designs as they are don't call for a "leave channel" option for an unsubscribed user. @pdpinch should we think about a new issue for this?